### ISSUE TITLE
[ERROR] Could not pass event InventoryClickEvent to Bookshelf v3.0.0.0

### ISSUE THREAD

**Issue by slooshied** [LOOHP/Bookshelf#3]
**How is it caused?**
It requires nothing. There were no players online when this error was recorded.

It should also be noted that this error happens repeatedly in the console, and spams it.

**Server Version:**
Paper Spigot Build #693

**Plugins:**
_(139) AdvancedAchievements, AdvancedDevelopmentLibrary, AdvancedNMotd, AdvancedRedstone, AdvancedTeleport, AlonsoTags, AreaShop, ArmorPlusPlus, ArmorStandEditor, ArtMap, AsyncWorldEdit, AutoPluginLoader, BalancedVillagerTrades, BetterCommandSpy, BettingGames, BigDoors, Bookshelf, Brewery, ButtonCooldown, Carz, ChatControl, ChestShop, Chunky, Citizens, ColoredAnvils*, ColoredBooked, CometWelcome, CommandBlocks, CommandHook, CommandSigns, CommandTimer, ConsoleSpamFix, ContainerPassthrough, Coordinates, CoreProtect, Criticals*, CustomCrafting, DeathMessages, DiscordSRV, DoubleOrNothing*, DoubleShulkerShells, DragonSlayer, dynmap*, DynmapHider*, EpicFarming, EpicFurnaces, EpicHoppers, Essentials, EssentialsChat, EssentialsSpawn, ExecutableItems, floodgate-bukkit, GCoreLegacy, Geyser-Spigot, GMusic, Graves, GroupManager, GSit, HackedCore*, HackedServer*, HeadDatabase, HiveChecker, HolographicDisplays, HolographicExtension, HolographicPlaceholders, ImageFireworksReborn, ImageOnMap, InfiniteDispensers, InfiniteFirework, InventoryRollback, InvUnload, ItemEdit, Jobs, JohnEnchanter*, KixsSimpleBackpacks, LibsDisguises, LWC, LWCTrust, mcMMO, MessageAnnouncer, MiniBoardGames, Morebosses, Multiverse-Core, MyCommand, NashornPlus, NBTAPI, NerfPhantoms, NexEngine, NoFlyZone, OfflineGrowth, OpenInv, Orebfuscator, PassiveMode, PermissionTimer, PlaceholderAPI, PlayerControlPlus, PlayerVaults, PotionExtender, PremiumVanish, ProtocolLib, QuantumShop, RideableEnderpearl, Roulette, Scoreboard-revision*, SelectionVisualizer, Shopkeepers, ShulkerBoxBackPacks, SimplePortals, SimpleRename, SkinsRestorer, SlotMachine, SpawnerMeta, SpecializedCrates, SpectatorMode, StackableItems, StoneCutterDamage, SurvivalInvisiframes, TAB, TabCompleteFilter, TempFly, TilePersistence, TobaccoPipes, Trade, UltimateAutoRestart, UltimateCatcher, UltimateTimber, UnbreakableEnchantment, VanillaEnchants*, Vault, VeinMiner, VillagerModifications, VillagerTradeLimiter, VoidSpawn, Voting, WITP, WolfyUtilities, WorldEdit, WorldGuard, WorldGuardExtraFlags_

```
[00:20:09 ERROR]: Could not pass event InventoryClickEvent to Bookshelf v3.0.0.0
java.lang.ClassCastException: class org.bukkit.block.DoubleChest cannot be cast to class com.loohp.bookshelf.objectholders.BookshelfHolder (org.bukkit.block.DoubleChest is in unnamed module of loader 'app'; com.loohp.bookshelf.objectholders.BookshelfHolder is in unnamed module of loader org.bukkit.plugin.java.PluginClassLoader @578f9c41)
        at com.loohp.bookshelf.listeners.BookshelfEvents.onUse(BookshelfEvents.java:362) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1440.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.5.jar:git-Paper-603]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.5.jar:git-Paper-603]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.5.jar:git-Paper-603]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:2700) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.PacketPlayInWindowClick.a(SourceFile:32) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:35) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1262) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1255) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1391) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1126) ~[patched_1.16.5.jar:git-Paper-603]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289) ~[patched_1.16.5.jar:git-Paper-603]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by LOOHP** [comment]
The issue should be already fixed in 3.0.0.1

**Comment by slooshied** [comment]
Didn't know a 3.0.0.1 had been released as I rely on Spigot's notifications when updating plugins. I was only able to find a build on the Dev jenkins. Thanks for the info however!!
